canonical-web-and-design/maas-ui
# Improve commissioning scripts selection UX

## Done

- Default commissioning scripts without the `noauto` tag, are selected by default.
- Default commissioning script tags are disabled (and have no X icon) as these scripts always run, regardless of whether they are selected in the UI.
- Adds a new `disabledTags: Tag[]` prop to `TagSelector`.
- Migrate `TagSelector` to TypeScript.
- Driveby fix for `act` warning from `ActionFormWrapper` test.

![image](https://user-images.githubusercontent.com/130286/90702489-7f53bc80-e2df-11ea-9236-445f51e799fd.png)
Default selection of disabled commissioning scripts.

![image](https://user-images.githubusercontent.com/130286/90702530-985c6d80-e2df-11ea-9597-efae324d109f.png)
All commissioning scripts added, hiding "select additional scripts" placeholder.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Commission a machine
- Ensure that default scripts are disabled, and that you can still add additional commissioning scripts.

- Add a script with `tag: noauto` via settings - commissioning scripts:

e.g.
```
#!/bin/bash
# --- Start MAAS 1.0 script metadata ---
# name: echo
# title: Runs echo 
# description: A useless script
# script_type: commissioning
# timeout: 60
# tags: noauto
# --- End MAAS 1.0 script metadata ---
echo 'hi'
```

- Return to the commissioning form and ensure the noauto script is not selected by default.

## Fixes

Fixes: #1488 

## Launchpad issue

[lp#1892238
](https://bugs.launchpad.net/maas/+bug/1892238)
## Backports

In general, please propose fixes against *master* rather than release branches (e.g. 2.7), unless the fix is only applicable for that specific release. Please apply backport labels to the PR (e.g. "Backport 2.7") for the appropriate releases to target.

Only bug and security fixes should be backported, new features should only land in master.
